### DISCUSSION TITLE
[理论] 开源治理的第一性原理

### DISCUSSION THREAD

**Discussion by will-ww** [kaiyuanshe/ONES#28] [00 开源治理系列参考规范整体设计]
**马斯洛需求层次理论**在现代行为科学中占有重要地位。马斯洛需求层次理论是**管理心理学**中人际关系理论、群体动力理论、权威理论、需求层次理论、社会测量理论的五大理论支柱之一。

马斯洛的需求层次结构包括人类需求的五级模型，通常被描绘成金字塔内的等级。从层次结构的底部向上，需求分别为：**生理**（食物和衣服），**安全**（工作保障），**社交需要**（友谊），**尊重**和**自我实现**。这种五阶段模式可分为不足需求和增长需求。前四个级别通常称为**缺陷需求**（D 需求），而最高级别称为**增长需求**（B 需求）。1943年马斯洛指出，人们需要动力实现某些需要，有些需求优先于其他需求。

<div align=center>
<img src="https://user-images.githubusercontent.com/15010826/158114250-ada223a8-5b4b-43a9-8858-5634f485d64a.png" width="500px">
</div>

开源和社区是一个广泛涉及到人和组织的话题，因此可以从管理心理学的视角来看待开源治理这件事情。实际上，很多企业管理、个人发展、公共关系、甚至国家治理等层面，都可以看到马斯洛需求层次结构的身影。

而国际上著名的 TODO 开源治理组织、以及欧盟的 OSPO 联盟等，也都参考了需求层次结构来设计开源治理的框架与能力模型。因此，我们的参考规范将马斯洛的需求层次结构作为开源治理的第一性原理，应该也是非常合适的。

<div align=center>
<img src="https://user-images.githubusercontent.com/15010826/158115393-30dc7c01-b7f2-4970-93f3-555bf2643768.png" width="500px">
</div>

同样也包括五个层次的内容：

- **数字战略**：全面拥抱与充分利用开源，通过开放式创新与开放组织，将开源作为数字化转型与数字主权的基石。
- **生态共建**：开源作为组织愿景的核心要素，通过主动回馈开源社区，成为开源生态建设与可持续发展的重要成员。
- **社区参与**：成为开源社区的一份子，建立并享受开源协作的模式，个体开始形成对开源社区的归属感。
- **安全可信**：安全可靠地使用开源技术，能够胜任对合规、依赖、漏洞等管理职责，组织建立起使用与管理开源的强大信心。
- **技术使用**：建立起开源技术的初步意识，知道如何高效地使用开源来创造价值，建立起使用开源的各项技能与经验积累。

这里我们先开个头，每个层次的内容还可以做大量的细化，然后更具规范的模板，就可以形成开源治理的参考规范了。而且还有一个好处是，由于普适性，这个五层模型可以应用到多个不同的场合，包括：企业端的开源治理、社区端的开源治理、政府和高校的开源治理等等，框架一样，只不过这些组织的目标和具体的实现细节不一样罢了。

**Comment by will-ww** [comment]
<div align=center>
<img src="https://user-images.githubusercontent.com/15010826/159430479-158e4762-6b07-4be5-b9d7-aca683ea5ec6.png" width="350px">
</div>

**1. 生理需求**
- 环境：舒适的环境，好椅子和咖啡！
- 工具：一台安静的电脑，具有良好的屏幕、鼠标和键盘。
- 流程：轻过程，迭代而不是瀑布
- 代码：避免技术债务，并且没有过时的框架

**2. 安全需求**
- 意义：你的工作有意义，产品会被使用
- 报酬：报酬反映工作
- 事业：有可能进步

**3. 社会归属**
- 愿景：项目愿景清晰
- 社区：属于一个社区，而不是单独的开发者
- 管理者：真正开始工作的管理者

**4. 尊重**
- 认可：被认可为有价值
- 学习：将工作经验添加到开发人员的个人资料中
- 贡献者：开发者为社区增加价值

**5. 自我实现**
- 责任：接受责任，负责
- 创造力：可以自由地创建您的解决方案

**Comment by wangleo61** [comment]
这个好

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

________________________________
发件人: Will Wang ***@***.***>
发送时间: Tuesday, March 22, 2022 3:35:54 PM
收件人: kaiyuanshe/ONES ***@***.***>
抄送: wangleo61 ***@***.***>; Manual ***@***.***>
主题: Re: [kaiyuanshe/ONES] [理论] 开源治理的第一性原理 (Discussion #28)


[image]<https://user-images.githubusercontent.com/15010826/159430479-158e4762-6b07-4be5-b9d7-aca683ea5ec6.png>

1. 生理需求

  *   环境：舒适的环境，好椅子和咖啡！
  *   工具：一台安静的电脑，具有良好的屏幕、鼠标和键盘。
  *   流程：轻过程，迭代而不是瀑布
  *   代码：避免技术债务，并且没有过时的框架

2. 安全需求

  *   意义：你的工作有意义，产品会被使用
  *   报酬：报酬反映工作
  *   事业：有可能进步

3. 社会归属

  *   愿景：项目愿景清晰
  *   社区：属于一个社区，而不是单独的开发者
  *   管理者：真正开始工作的管理者

4. 尊重

  *   认可：被认可为有价值
  *   学习：将工作经验添加到开发人员的个人资料中
  *   贡献者：开发者为社区增加价值

5. 自我实现

  *   责任：接受责任，负责
  *   创造力：可以自由地创建您的解决方案

―

**Comment by wangleo61** [comment]
我现在处理努力的第一阶段😊

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

发件人: Will ***@***.***>
发送时间: Tuesday, March 22, 2022 3:36 PM
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [kaiyuanshe/ONES] [理论] 开源治理的第一性原理 (Discussion #28)


[image]<https://user-images.githubusercontent.com/15010826/159430479-158e4762-6b07-4be5-b9d7-aca683ea5ec6.png>

1. 生理需求

  *   环境：舒适的环境，好椅子和咖啡！
  *   工具：一台安静的电脑，具有良好的屏幕、鼠标和键盘。
  *   流程：轻过程，迭代而不是瀑布
  *   代码：避免技术债务，并且没有过时的框架

2. 安全需求

  *   意义：你的工作有意义，产品会被使用
  *   报酬：报酬反映工作
  *   事业：有可能进步

3. 社会归属

  *   愿景：项目愿景清晰
  *   社区：属于一个社区，而不是单独的开发者
  *   管理者：真正开始工作的管理者

4. 尊重

  *   认可：被认可为有价值
  *   学习：将工作经验添加到开发人员的个人资料中
  *   贡献者：开发者为社区增加价值

5. 自我实现

  *   责任：接受责任，负责
  *   创造力：可以自由地创建您的解决方案

**Comment by frank-zsy** [comment]
个人感觉目前的框架中，最底层的一层其实和上层的多个层次是有耦合的，例如监控开源使用、物料清单其实和可信有重叠，发展路线制定和生态甚至战略略有重叠。又或者安全可信其实是一个纵向的层面，在参与、主导、生态、战略层面可能也都有安全可信的诉求。

**Reply by will-ww** [reply]
> 个人感觉目前的框架中，最底层的一层其实和上层的多个层次是有耦合的，例如监控开源使用、物料清单其实和可信有重叠，发展路线制定和生态甚至战略略有重叠。又或者安全可信其实是一个纵向的层面，在参与、主导、生态、战略层面可能也都有安全可信的诉求。

对滴，所以怎样结构化，感觉是个挑战。层次结构有时候也会有误解。目前想表达的是目标，一个组织在做开源治理的时候，肯定会同时有几个目标组合，特别是最底下的两层，有普遍的目标需求。

**Comment by wangleo61** [comment]
认同Frank的观点哈

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

发件人: Frank ***@***.***>
发送时间: Sunday, March 27, 2022 10:15 PM
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [kaiyuanshe/ONES] [理论] 开源治理的第一性原理 (Discussion #28)


个人感觉目前的框架中，最底层的一层其实和上层的多个层次是有耦合的，例如监控开源使用、物料清单其实和可信有重叠，发展路线制定和生态甚至战略略有重叠。又或者安全可信其实是一个纵向的层面，在参与、主导、生态、战略层面可能也都有安全可信的诉求。

―

**Comment by gymgym1212** [comment]
在生理需求——技术使用这一层次中，我觉得:

1. 开源教育。对组织内部的所有人员进行开源教育，包括开源历史、定义、许可证、治理、最佳实践，还包括相关工具的使用等等。通过一系列持续的教育工作，让所有人了解开源，知道如何有礼貌的参与。

2. 开源治理。企业、基金会、社区，都需要成立核心的开源治理小组（在不同组织下这个小组的称呼可能不尽相同），负责对组织内部的开源活动进行指引和管理，包括上述的开源教育活动。另外，一个组织的形象是统一的。核心开源治理小组需要作为所有开源活动出入口的把关者，结合自身情况（组织目标、国情、文化等）去制定相应的开源治理规范，包括怎么用别人的开源项目、怎么参与别人的开源项目、怎么发布自己的开源项目、自己的开源项目如何让别人参与等。

3. 开源工具。开源治理小组注定是一个小的单位，所以还需要配合各种工具去提高治理效能，比如：开源看板、代码审查工具等。当然这一块其实也涉及到更高层次的`安全`需求，也恰好和 Frank 、永雷老师说的一样，我也认为这是一个纵向的，生长的，从弱到强的过程。
